core/descriptors/src/org/jetbrains/kotlin/types/checker/ClassicTypeSystemContext.kt

```diff
@@ -719,13 +719,13 @@ interface ClassicTypeSystemContext : TypeSystemInferenceExtensionContext, TypeSy
     }
 
     override fun TypeParameterMarker.getName(): Name {
-        require(this is TypeParameterDescriptor, this::errorMessage)
-        return name
+        require(this is AbstractTypeParameterDescriptor.TypeParameterTypeConstructor, this::errorMessage)
```

Comment:
Is this a breaking change for other classes implementing `TypeParameterMarker`? How about converting `AbstractTypeParameterDescriptor` to kotlin and make `TypeParameterTypeConstructor` implementing `TypeParameterDescriptor` through delegation?

Reply:
Indeed. I knew it will break, but just curious how seriously it would, and boom! 9365 failures. ;)

I tried a manual delegation, and then gave up, and somehow thought it's not allowed to convert these old descriptors to `.kt`. I don't know why. Thanks for the suggestion.